src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -118,6 +118,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     
   </Target>
 
+  <Target Name="_CheckForMismatchingPlatform"
+          BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
```

Comment:
When targeting .NET Framework, we default to the current architecture if it's not set, but then later back out that change.  Can you see how that interacts with this check, and add a test that covers it?